climpyrical/gridding.py

```diff
@@ -0,0 +1,535 @@
+import numpy as np
+from scipy.interpolate import NearestNDInterpolator
+from pyproj import Transformer, Proj
+
+
+def check_ndims(data, n):
+    """Checks that a provided array has n dimensions
+    Args:
+        data (np.ndarray): Array to check dimensions of
+        n (integer): data's expected dimensions
+    Raises:
+        TypeError:
+            If data or n are not arrays or an integer
+        ValueError:
+            If data's dimension is not expected
+    """
+    if not isinstance(data, np.ndarray):
```

Comment:
**This is not a change request**, just a bit of advice: 

Whilst checking the types of all your arguments is admirable, it is also laborious, time consuming, and can be done better with the new-ish [type annotations](https://docs.python.org/3/library/typing.html). If something else would burk (a.k.a. raise an exception) for wrong-type args, then don't bother type checking ...

Unless it is utterly central to your code and there is a distinct possibility of it being fooled. Or unless these functions will be exposed for wide use by others. Then you  want more careful checking and very specific error messages.

Reply:
I mainly included them to guarantee that the follow-up attributes I need to check, i.e. `max()`, `min()`, `size`, and some of the functions like `np.any` or `np.isnan` will definitely work with any numpy.narray provided. 

I agree, typing is the way forward. Down the road, look for future PRs that include that. 

I know there are some objects other than `np` that share some of these attributes, but I'm not sure if `climpyrical` should have multiple object compatibility just yet. I do see that as a nice feature down the road.

Reply:
Read that as: I'm not sure if `numpy.ndarrays` are even the best data type for `climpyrical` to pass around. I am looking into perhaps using just `xarray` objects or `dask` objects in the future.

Reply:
That's a much bigger discussion.